File: src/Leibniz.Client/src/app/common/domain/result-set.ts
import { EntityType } from "../../relationships/domain/entity-type";

export interface ResultSet<T> {
  index: number;
  limit: number;
  count: number;
  query?: string;
  type?: EntityType;
  id?: number;
  filterType?: EntityType;
  filterId?: number;
  data: T[];
}

File: src/Leibniz.Api/Common/ResultSet.cs
namespace Leibniz.Api.Common;
public record ResultSet<T>
{
    public long Index { get; init; }
    public long Limit { get; init; }
    public long Count { get; init; }
    public List<T> Data { get; init; }
    public string? Query { get; init; }
    public EntityType? Type { get; init; }
    public long? Id { get; init; }
    public EntityType? FilterType { get; init; }
    public long? FilterId { get; init; }
    public bool? IsPrimary { get; init; }
}
